A more idiomatic Emacs's approach (#3064)
ocaml/ocaml.org: data/tutorials/getting-started/2_00_editor_setup.md

    1. Go to `File` > `Preferences` > `Settings` view (or press `Ctrl ,`)
    2. Select `User` > `Extensions` > `OCaml Platform`
    3. Uncheck `OCaml: Use OCaml Env`. That's it!
    
## Emacs

> Using Emacs as the main OCaml editor requires at least two modes:

Reviewer: ```suggestion
Using Emacs to work with OCaml requires at least two modes:
```

- A major mode, which, among other things, supports syntax highlighting and the structuring of indentation levels

> - A minor mode, which will interact with a server language (such as `ocaml-lsp-server` or `merlin`). In this tutorial, we will focus on using `ocaml-lsp-server` as a server.

Reviewer: ```suggestion
- A minor mode, which will interact with a language server (such as `ocaml-lsp-server` or `merlin`). In this tutorial, we will focus on using the new `ocaml-eglot` mode and `ocaml-lsp-server` as a server.
```

### Choosing a major mode

> There are several major modes dedicated to OCaml, although there are 3 main ones:

Reviewer: ```suggestion
There are several major modes dedicated to OCaml, of which the 3 main ones are:
```

> - [Tuareg](https://github.com/ocaml/tuareg): an old-fashioned (but still updated), very complete method, often recommended

Reviewer: ```suggestion
- [Tuareg](https://github.com/ocaml/tuareg): an old-fashioned (but still updated), very complete mode, usually the recommended one
```

- [Caml](https://github.com/ocaml/caml-mode): a mode even older than `tuareg` (but still updated), lighter than `tuareg`

> - [Neocaml](https://github.com/bbatsov/neocaml): a brand new mode, based on modern tools (like [tree-sitter](https://tree-sitter.github.io/tree-sitter/)). Currently still very experimental!

Reviewer: ```suggestion
- [Neocaml](https://github.com/bbatsov/neocaml): a brand new mode, based on modern tools (like [tree-sitter](https://tree-sitter.github.io/tree-sitter/)). Still experimental at the time of writing.
```

> For the purposes of this tutorial, we're going to focus on the use of `tuareg` as the main mode, but that's a detail and you're free to experiment and choose your favourite mode! You can add these lines to your Emacs configuration:

Reviewer: ```suggestion
For the purposes of this tutorial, we are going to focus on the use of `tuareg` as the major mode, but you should feel free to experiment and choose your favourite one! To use `tuareg`, you can add these lines to your Emacs configuration:
```

```elisp
(use-package tuareg
  :ensure t
  :mode (("\\.ocamlinit\\'" . tuareg-mode)))
```

> > If your version of Emacs does not support the `use-package` macro (or is not set up to take MELPA packages into account), please update it and follow these instructions for [`use-package`](https://github.com/jwiegley/use-package) and to [add MELPA](https://melpa.org/#/getting-started).

Reviewer: ```suggestion
> If your version of Emacs does not support the `use-package` macro (or is not set up to take MELPA packages into account), please update it and follow these instructions to install [`use-package`](https://github.com/jwiegley/use-package) and [MELPA](https://melpa.org/#/getting-started).
```

> > If your version of Emacs does not support the `use-package` macro (or is not set up to take MELPA packages into account), please update it and follow these instructions to install [`use-package`](https://github.com/jwiegley/use-package) and [MELPA](https://melpa.org/#/getting-started).

Reviewer 2: Quotes should be used for quotes, not highlighting. We usually have paragraph header, in bold, such as “remark”, “fun fact”, “warning”, “note” or something else.

Author: Should be fixed on https://github.com/ocaml/ocaml.org/pull/3064/commits/c1bb0be2592f67472a775588cf7a528ca23bd561

### LSP setup for OCaml

> Since version `29.1`, Emacs has had a mode for interacting with an LSP server, [Eglot](https://www.gnu.org/software/emacs/manual/html_mono/eglot.html). If you are using an earlier version of Emacs, you will need to install it this way:

Reviewer: ```suggestion
Since version `29.1`, Emacs has had a built-in mode for interacting with LSP servers, [Eglot](https://www.gnu.org/software/emacs/manual/html_mono/eglot.html). If you are using an earlier version of Emacs, you will need to install it this way:
```

```elisp
(use-package eglot
  :ensure t)
```

Next, we need to bridge the gap between our major mode (in this case, `tuareg`) and `eglot`. This is done using the [`ocaml-eglot`](https://github.com/tarides/ocaml-eglot) package:

```elisp
(use-package ocaml-eglot
  :ensure t
  :after tuareg
  :hook
  (tuareg-mode . ocaml-eglot)
  (ocaml-eglot . eglot-ensure))
```

> And that's all there is to it! Now all we need to do is install `ocaml-lsp-server` and `ocamlformat` in our [switch](/docs/opam-switch-introduction), exactly as we did for Visual Studio Code:

Reviewer: ```suggestion
And that's all there is to it! Now all you need to do is install `ocaml-lsp-server` and `ocamlformat` in our [switch](/docs/opam-switch-introduction):
```

```shell
opam install ocaml-lsp-server ocamlformat
```

> And you're ready to edit OCaml code _productively_ with Emacs!

Reviewer: ```suggestion
You are now ready to edit OCaml code _productively_ with Emacs!
```

> > OCaml-eglot can be finely configured, the project [README](https://github.com/tarides/ocaml-eglot/blob/main/README.md) gives several configuration paths to adapt perfectly to your usual workflow! You'll also find an exhaustive presentation of the different functions offered by the mode!

Reviewer: ```suggestion
> OCaml-eglot can be finely configured, the project [README](https://github.com/tarides/ocaml-eglot/blob/main/README.md) gives several configuration paths to adapt perfectly to your workflow. You will also find there an exhaustive presentation of the different functions offered by the mode.
```

> > OCaml-eglot can be finely configured, the project [README](https://github.com/tarides/ocaml-eglot/blob/main/README.md) gives several configuration paths to adapt perfectly to your workflow. You will also find there an exhaustive presentation of the different functions offered by the mode.

Reviewer 2: Same as above

Author: Should be fixed on https://github.com/ocaml/ocaml.org/pull/3064/commits/c1bb0be2592f67472a775588cf7a528ca23bd561
